types.go

```diff
@@ -97,55 +97,23 @@ type Stats struct {
 	Runtime         time.Duration // Total runtime of the optimization
 }
 
-// FunctionInfo is data to give to the optimizer about the objective function.
-type FunctionInfo struct {
-	IsGradient                bool
-	IsHessian                 bool
-	IsFunctionGradient        bool
-	IsFunctionGradientHessian bool
-	IsStatuser                bool
+// ProblemInfo is data to give to the optimizer about the objective function.
+type ProblemInfo struct {
+	HasGradient bool
+	HasHessian  bool
+	HasStatus   bool
```

Comment:
I think that we can also remove `HasStatus`, it is a piece of information that is useless to `Method`.